sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -407,8 +407,17 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
 
   override def catalogString: String = {
```

Comment:
```scala
    val stringConcat = new StringUtils.StringConcat()
    val len = fields.length
    stringConcat.append("struct<")
    var i = 0
    while (i < len) {
      stringConcat.append(s"${fields(i).name}:${fields(i).dataType.catalogString}")
      i += 1
      if (i < len) stringConcat.append(",")
    }
    stringConcat.append(">")
    stringConcat.toString
```

Reply:
Thanks! I wan wanting to use `init` to make it looks simple. Now seems not simpler than while loop...